presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -69,6 +70,11 @@
     private final boolean verboseStats;
     private final boolean cpuTimerEnabled;
 
+    private final Lock memoryTimeIntegralLock = new ReentrantLock();
```

Comment:
The way you use it. You would just want an `Object` instead.

`private final Object memoryTimeIntegralLock = new Object();`

You never use `Lock` as a monitor.